### PR TITLE
OBPIH-3658 Add invoice data model

### PR DESCRIPTION
**Pull request by awalkowiak**

### CODE REVIEW

File: grails-app/domain/org/pih/warehouse/invoice/InvoiceItem.groovy [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    def beforeUpdate = {
+        def currentUser = AuthService.currentUser.get()
+        if (currentUser) {
+            updatedBy = currentUser
+        }
+    }
+
+    String id
+
+    Invoice invoice
```
Comment (jmiranda) [inline grails-app/domain/org/pih/warehouse/invoice/InvoiceItem.groovy:38]: This should be a 
```
static belongsTo = [invoice: Invoice]
```

File: grails-app/domain/org/pih/warehouse/invoice/Invoice.groovy [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    def beforeUpdate = {
+        def currentUser = AuthService.currentUser.get()
+        if (currentUser) {
+            updatedBy = currentUser
+        }
+    }
+
+    String id
+    String invoiceNumber
+    String referenceNumber
```
Comment (jmiranda) [inline grails-app/domain/org/pih/warehouse/invoice/Invoice.groovy:36]: Can you remind me what this is for?

Comment (awalkowiak) [reply grails-app/domain/org/pih/warehouse/invoice/Invoice.groovy:36]: @jmiranda Should I remove it and leave only the invoiceNumber?

Comment (jmiranda) [reply grails-app/domain/org/pih/warehouse/invoice/Invoice.groovy:36]: No just add a hasMany for referenceNumber like shipment.
```
static hasMany = [referenceNumbers: ReferenceNumber]
```

**Comment by awalkowiak** [comment]
@jmiranda I pushed requested changes in a separate commit for easier review.

**Review by jmiranda — APPROVED** [review]

**Pull request merged**